subsys/net/ip/net_context.c

```diff
@@ -759,6 +760,11 @@ int net_context_connect(struct net_context *context,
 		return -EINVAL; /* Not IPv4 or IPv6 */
 	}
 
+	ret = net_tls_connect(context, false);
```

Comment:
Shouldn't `net_tls_connect` be called after `net_tcp_connect`, as it is supposed to do the TLS handshake?